pajbot/config.py

```diff
@@ -1,20 +1,117 @@
+        return config["main"]["streamer_id"], None
+    elif "streamer" in config["main"]:
+        log.warning(
+            "DEPRECATED - Specify the streamer's Twitch User ID using streamer_id instead of their login name in streamer"
+        )
+        return None, config["main"]["streamer"]
     elif "target" in config["main"]:
-        channel = config["main"]["target"]
-        return f"{channel[1:]}", channel
+        log.warning("DEPRECATED - You should specify the streamer using the streamer_id config value")
+        return None, config["main"]["target"][1:]
+
+    raise KeyError("Missing streamer_id key from config")
+
+
+def load_streamer(config: Config, twitch_helix_api: TwitchHelixAPI) -> UserBasics:
+    streamer_id, streamer_login = load_streamer_id_or_login(config)
+    if streamer_id is not None:
+        return twitch_helix_api.require_user_basics_by_id(streamer_id)
+    if streamer_login is not None:
+        return twitch_helix_api.require_user_basics_by_login(streamer_login)
+    raise ValueError("Bad config, missing streamer id or login")
+
+
+def load_bot_id_or_login(config: Config) -> Union[Tuple[str, None], Tuple[None, str]]:
+    """
+    Load either the bot Twitch User ID or Twitch User Login from the config
+    The Twitch User ID is read from bot_id, and the Twitch User Login us read from nickname
+
+    If neither of the values are present, throw an exception
+    """
+
+    if "bot_id" in config["main"]:
+        return config["main"]["bot_id"], None
+    elif "nickname" in config["main"]:
+        log.warning(
+            "DEPRECATED - Specify the bot's Twitch User ID using bot_id instead of their login name in nickname"
+        )
+        return None, config["main"]["nickname"]
+
+    raise KeyError("Missing bot_id key from config")
+
+
+def load_bot(config: Config, twitch_helix_api: TwitchHelixAPI) -> UserBasics:
+    bot_id, bot_login = load_bot_id_or_login(config)
+    if bot_id is not None:
+        return twitch_helix_api.require_user_basics_by_id(bot_id)
+    if bot_login is not None:
+        return twitch_helix_api.require_user_basics_by_login(bot_login)
+    raise ValueError("Bad config, missing bot id or login")
+
+
+def load_control_hub_id_or_login(config: Config) -> Union[Tuple[str, None], Tuple[None, str]]:
+    """
+    Load either the control hub Twitch User ID or Twitch User Login from the config
+    The Twitch User ID is read from control_hub_id, and the Twitch User Login us read from control_hub
+
+    If neither of the values are present, throw an exception
+    """
+
+    if "control_hub_id" in config["main"]:
+        return config["main"]["control_hub_id"], None
+    elif "control_hub" in config["main"]:
+        log.warning(
+            "DEPRECATED - Specify the control hub's Twitch User ID using control_hub_id instead of their login name in control_hub"
+        )
+        return None, config["main"]["control_hub"]
+
+    raise KeyError("Missing control_hub_id key from config")
```

Comment:
This wasn't a required field before. Certain channels I don't have a control hub set and I don't think it should be required.

Reply:
Fixed in [`5e8f8e3` (#1590)](https://github.com/pajbot/pajbot/pull/1590/commits/5e8f8e335eeafa6bd1aa125f956bbf443c4c44b7)